File: knuth_java/src/test/java/com/chapter1/GcdTest.java
package com.chapter1;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class GcdTest {

    @Test
    void calculate() {
        assertEquals(1, new Gcd(7, 5).calculate());
        assertEquals(6, new Gcd(18, 6).calculate());
        assertEquals(57, new Gcd(2166, 6099).calculate());
    }
}

File: knuth_kotlin/src/test/kotlin/chapter1/GcdTest.kt
package chapter1

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class GcdTestKotlin {
    @Test
    fun testGcd() {
        assertEquals(1, gcd(7, 5))
        assertEquals(6, gcd(18, 6))
        assertEquals(57, gcd(2166, 6099))
    }
}